frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -158,37 +169,55 @@ export const insightLogic = kea<insightLogicType>({
                 setTimeout(() => {
                     if (values && view == values.activeView) {
                         actions.setShowTimeoutMessage(true)
+                        const tags = {
+                            insight: values.activeView,
+                            scene: sceneLogic.values.scene,
+                        }
+                        posthog.capture('insight timeout message shown', tags)
+                        captureInternalMetric({ method: 'incr', metric: 'insight_timeout', value: 1, tags })
                     }
                 }, SHOW_TIMEOUT_MESSAGE_AFTER)
             )
             actions.setIsLoading(true)
         },
-        endQuery: ({ view, lastRefresh, exception }) => {
-            clearTimeout(values.timeout || undefined)
+        endQuery: ({ queryId, view, lastRefresh, exception }) => {
+            if (values.timeout) {
+                clearTimeout(values.timeout)
+            }
             if (view === values.activeView) {
                 actions.setShowTimeoutMessage(values.maybeShowTimeoutMessage)
                 actions.setShowErrorMessage(values.maybeShowErrorMessage)
                 actions.setLastRefresh(lastRefresh || null)
                 actions.setIsLoading(false)
-                if (values.maybeShowTimeoutMessage) {
-                    posthog.capture('insight timeout message shown', { insight: values.activeView, ...exception })
+
+                const duration = new Date().getTime() - values.queryStartTimes[queryId]
+                const tags = {
+                    insight: values.activeView,
+                    scene: sceneLogic.values.scene,
+                    success: !exception,
+                    ...exception,
                 }
+
+                posthog.capture('insight loaded', { ...tags, duration })
+                captureInternalMetric({ method: 'timing', metric: 'insight_load_time', value: duration, tags })
```

Comment:
Looks like we're using the same load time metric for success and failure (assuming from the maybeShowErrorMessage below that failures also come here), I wonder if it would make sense separating these, because otherwise fast failing load times can make us draw wrong conclusions. Potentially we can filter in team.py later to only look at load_time for succeeding insights.

Reply:
We do have a property for success or not. I don't have strong feelings here, I think either approach works and it's more important to iterate fast

Reply:
Just to confirm that I've understood things correctly currently load_time is calculated on top of both succeeding and failing loadings? 
If no, then I assume we already filter load_time by success and I failed at reading the code.
If yes, then did you mean by your question, that either approach (changing here or changing the filtering in the dashboard later) is fine, if yes, I agree and my main point was that it would be good to have load time only be from successful loads & it seemed like it wasn't from here nor filtered later (again I might have misunderstood).

Reply:
Will update the dashboard query in a follow-up PR :)